utils/test/e2e/shared_steps.py

```diff
@@ -70,14 +93,30 @@ def pipeline_has_finished_with_state(context, pipeline_name: str, state: str):
 @then(
     "the files {output_files} are present in the ADLS container {container_name} in the directory " "{directory_name}"
 )
-def check_all_files_present_in_adls(context, output_files, container_name, directory_name):
+def check_all_files_present_in_adls(context, output_files: str, container_name: str, directory_name: str):
+    """
+    Check if all the specified files are present in the specified ADLS container and directory.
+
+    Args:
+        context: Behave context object.
+        output_files: JSON-encoded list of expected file names.
+        container_name: The name of the ADLS container.
+        directory_name: The directory path in the container.
+    """
     expected_files_list = json.loads(output_files)
     test_directory_name = f"{directory_name}/automated_tests/{context.test_run_id}"
     assert all_files_present_in_adls(adls_client, container_name, test_directory_name, expected_files_list)
 
 
 @step("the ADF pipeline completed in less than {seconds} seconds")
-def check_adf_pipeline_completion_time(context, seconds):
+def check_adf_pipeline_completion_time(context, seconds: str):
```

Comment:
add type annotation for context? (here and in other places)

Reply:
done